catalog/spatial/wfs/spatial-wfs-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/converter/impl/AbstractFeatureConverter.java

```diff
@@ -139,114 +136,137 @@ protected HierarchicalStreamReader copyXml(
-            String bytes = convertToBytes(reader, metacardMapper.getDataUnit());
-            if (StringUtils.isNotBlank(bytes)) {
-              LOGGER.debug(
-                  "Setting mapped metacard attribute {} with value {}",
-                  mappedMetacardAttribute,
-                  bytes);
-              mc.setAttribute(mappedMetacardAttribute, bytes);
-            }
-            if (StringUtils.isNotBlank(sizeBeforeConversion)) {
-              LOGGER.debug(
-                  "Setting metacard attribute {} with value {}",
-                  featureProperty,
-                  sizeBeforeConversion);
-              mc.setAttribute(featureProperty, sizeBeforeConversion);
-            }
-          } else {
-            value =
-                getValueForMetacardAttribute(
-                    attributeDescriptor.getType().getAttributeFormat(), reader);
-            if (value != null) {
-              LOGGER.debug(
-                  "Setting mapped metacard attribute {} with value {}",
-                  mappedMetacardAttribute,
-                  value);
-              mc.setAttribute(mappedMetacardAttribute, value);
-              mc.setAttribute(featureProperty, value);
-            }
-          }
-        } else {
-          value =
-              getValueForMetacardAttribute(
-                  attributeDescriptor.getType().getAttributeFormat(), reader);
-
-          if (value != null) {
-            LOGGER.debug("Setting metacard attribute {} with value {}", featureProperty, value);
-            mc.setAttribute(featureProperty, value);
-          }
-        }
-        if (BasicTypes.GEO_TYPE
+    if ((featureValue != null && StringUtils.isNotBlank(featureValue.toString()))
+        || BasicTypes.GEO_TYPE
+            .getAttributeFormat()
+            .equals(attributeDescriptor.getType().getAttributeFormat())
+        || BasicTypes.DATE_TYPE
             .getAttributeFormat()
             .equals(attributeDescriptor.getType().getAttributeFormat())) {
-          mc.setLocation((String) value);
-        }
-        // if this node matches a basic metacard attribute name,
-        // populate that field as well
-        if (isBasicMetacardAttribute(reader.getNodeName())) {
-          LOGGER.debug("Setting metacard basic attribute: {} = {}", reader.getNodeName(), value);
 
-          mc.setAttribute(reader.getNodeName(), value);
+      // do this before applying the template
+      if (BasicTypes.GEO_TYPE
+          .getAttributeFormat()
+          .equals(attributeDescriptor.getType().getAttributeFormat())) {
+        Serializable location = context.get(entry.getFeatureProperty());
+
+        if (location != null) {
+          metacard.setLocation(location.toString());
         }
       }
 
-      reader.moveUp();
+      Serializable attributeValue = getAttributeValue(attributeDescriptor, featureValue);
+      metacard.setAttribute(attributeDescriptor.getName(), attributeValue);
```

Comment:
i supposed we can conditionally set the attribute to make sonarqube happy; though i think setting it to null ultimately has no negative effect